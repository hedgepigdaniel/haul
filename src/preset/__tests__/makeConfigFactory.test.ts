import webpack from 'webpack';
import path from 'path';
import makeConfigFactory from '../makeConfigFactory';
import { Runtime, ProjectConfig, EnvOptions } from '../../';
// @ts-ignore
import { replacePathsInObject } from 'jest/helpers'; // eslint-disable-line

const makeConfig = makeConfigFactory(
  (_runtime, env, bundleName, projectConfig) => ({
    mode: env.dev ? 'development' : 'production',
    context: env.root,
    devtool: false,
    entry: projectConfig.bundles[bundleName].entry,
    target: 'webworker',
    output: {
      path: env.assetsDest || env.root,
      publicPath: `http://${projectConfig.server.host}:${projectConfig.server.port}/`,
      globalObject: 'this',
    },
    plugins: [
      new webpack.DefinePlugin({
        'process.env': {
          NODE_ENV: env.dev ? '"development"' : '"production"',
        },
        __DEV__: env.dev,
      }),
      new webpack.LoaderOptionsPlugin({
        minimize: !!env.minify,
        debug: env.dev,
      }),
    ],
  })
);

function hasPlugin(webpackConfig: any, pluginName: string) {
  return webpackConfig.plugins.some(
    (p: any) => p.constructor.name === pluginName
  );
}

const runtime = new Runtime();

describe('makeConfig', () => {
  describe('with single bundle', () => {
    const baseEnv: EnvOptions = {
      platform: 'ios',
      root: __dirname,
      dev: false,
      bundleMode: 'single-bundle',
    };

    it('should create config for basic-bundle', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(replacePathsInObject(config)).toMatchSnapshot();
    });

    it('should create config with multiple entry files', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: ['./a.js', './b.js'],
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(config.webpackConfigs.index.entry).toEqual(['./a.js', './b.js']);
    });

    it('should create config with builder function and custom transform', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: (env, runtime) => {
            expect(env.bundleType).toEqual('basic-bundle');
            expect(runtime).toBeDefined();

            return {
              entry: 'index.js',
              transform({ bundleName, config, runtime, env }) {
                expect(bundleName).toEqual('index');
                expect(runtime).toBeDefined();
                expect(env).toBeDefined();

                config.stats = 'minimal';
              },
            };
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(config.webpackConfigs.index.stats).toEqual('minimal');
    });

    it('should create config for indexed-ram-bundle', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'indexed-ram-bundle',
        bundleTarget: 'file',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackRamBundlePlugin')
      ).toBeTruthy();
    });

    it('should create config for file-ram-bundle', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'file-ram-bundle',
        bundleTarget: 'file',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackRamBundlePlugin')
      ).toBeTruthy();
    });

    it('should not create config with external source maps if sourceMap=false', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            sourceMap: false,
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'SourceMapDevToolPlugin')
      ).toBeFalsy();
    });

    it('should create config with inline source maps if sourceMap="inline"', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            sourceMap: 'inline',
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'EvalSourceMapDevToolPlugin')
      ).toBeTruthy();
    });

    it('should create config for packager server', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const runtime = new Runtime();
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'indexed-ram-bundle',
        bundleTarget: 'server',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackRamBundlePlugin')
      ).toBeFalsy();
    });

    it('should use filename from bundleOutput instead of templates', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
        bundleOutput: path.join(__dirname, 'main.jsbundle'),
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(config.webpackConfigs.index.output!.filename).toEqual(
        'main.jsbundle'
      );
    });

    it('should allow to overwrite bundle name', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            name: 'main',
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(config.webpackConfigs.main.output!.filename).toEqual(
        'main.jsbundle'
      );
    });

    it('should use custom template', () => {
      const projectConfig: ProjectConfig = {
        templates: {
          filename: {
            ios: '[bundleName].[platform].[type].[mode].jsbundle',
          },
        },
        bundles: {
          main: {
            entry: 'index.js',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleType: 'basic-bundle',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(config.webpackConfigs.main.output!.filename).toEqual(
        'main.ios.default.prod.jsbundle'
      );
    });
  });

  describe('with multiple bundles', () => {
    const baseEnv: EnvOptions = {
      platform: 'ios',
      root: __dirname,
      dev: false,
      bundleMode: 'multi-bundle',
      bundleTarget: 'file',
    };

    it('should create config', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: './host.js',
            dependsOn: ['base_dll'],
          },
          base_dll: {
            entry: './dll.js',
            dll: true,
            type: 'indexed-ram-bundle',
          },
          app: {
            entry: './spp.js',
            app: true,
            dependsOn: ['base_dll'],
            type: 'indexed-ram-bundle',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleOutput: __dirname,
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.base_dll, 'WebpackRamBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.app, 'WebpackRamBundlePlugin')
      ).toBeTruthy();
      expect(replacePathsInObject(config)).toMatchSnapshot();
    });

    it('should create config for packager server', () => {
      const projectConfig: ProjectConfig = {
        bundles: {
          index: {
            entry: './host.js',
            dependsOn: ['base_dll'],
          },
          base_dll: {
            entry: './dll.js',
            dll: true,
            type: 'indexed-ram-bundle',
          },
          app: {
            entry: './spp.js',
            app: true,
            dependsOn: ['base_dll'],
            type: 'indexed-ram-bundle',
          },
        },
      };
      const env: EnvOptions = {
        ...baseEnv,
        bundleTarget: 'server',
      };
      const config = makeConfig(projectConfig)(runtime, env);
      expect(
        hasPlugin(config.webpackConfigs.index, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.base_dll, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(
        hasPlugin(config.webpackConfigs.app, 'WebpackBasicBundlePlugin')
      ).toBeTruthy();
      expect(replacePathsInObject(config)).toMatchSnapshot();
    });
  });
});
