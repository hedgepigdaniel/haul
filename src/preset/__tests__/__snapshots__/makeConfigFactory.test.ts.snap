// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeConfig with multiple bundles should create config 1`] = `
Object {
  "bundles": Object {
    "app": Object {
      "app": true,
      "assetsDest": "",
      "dependsOn": Array [
        "base_dll",
      ],
      "dev": false,
      "dll": false,
      "entry": Array [
        "./spp.js",
      ],
      "external": false,
      "hasteOptions": Object {},
      "minify": false,
      "minifyOptions": undefined,
      "name": "app",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "indexed-ram-bundle",
    },
    "base_dll": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [],
      "dev": false,
      "dll": true,
      "entry": Array [
        "./dll.js",
      ],
      "external": false,
      "hasteOptions": Object {},
      "minify": false,
      "minifyOptions": undefined,
      "name": "base_dll",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "indexed-ram-bundle",
    },
    "index": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [
        "base_dll",
      ],
      "dev": false,
      "dll": false,
      "entry": Array [
        "./host.js",
      ],
      "external": false,
      "hasteOptions": Object {},
      "minify": false,
      "minifyOptions": undefined,
      "name": "index",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
  },
  "platforms": Array [
    "ios",
    "android",
  ],
  "server": Object {
    "host": "localhost",
    "port": 8081,
  },
  "templates": Object {
    "filename": Object {
      "__server__": "[bundleName].[platform].bundle",
      "android": "[bundleName].[platform].bundle",
      "ios": "[bundleName].jsbundle",
    },
  },
  "webpackConfigs": Object {
    "app": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Array [
        "./spp.js",
      ],
      "mode": "production",
      "output": Object {
        "filename": "app.jsbundle",
        "globalObject": "this",
        "library": "app",
        "libraryTarget": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackRamBundlePlugin {
          "bundleName": "index",
          "indexRamBundle": true,
          "minify": false,
          "minifyOptions": undefined,
          "modules": Array [],
          "name": "WebpackRamBundlePlugin",
          "preloadBundles": Array [
            "base_dll",
          ],
          "singleBundleMode": false,
          "sourceMap": true,
        },
        DllReferencePlugin {
          "options": Object {
            "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
            "manifest": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
            "sourceType": "this",
          },
        },
      ],
      "target": "webworker",
    },
    "base_dll": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Array [
        "./dll.js",
      ],
      "mode": "production",
      "output": Object {
        "filename": "base_dll.jsbundle",
        "globalObject": "this",
        "library": "base_dll",
        "libraryTarget": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackRamBundlePlugin {
          "bundleName": "index",
          "indexRamBundle": true,
          "minify": false,
          "minifyOptions": undefined,
          "modules": Array [],
          "name": "WebpackRamBundlePlugin",
          "preloadBundles": Array [],
          "singleBundleMode": false,
          "sourceMap": true,
        },
        DllPlugin {
          "options": Object {
            "name": "base_dll",
            "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
          },
        },
      ],
      "target": "webworker",
    },
    "index": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Array [
        "./host.js",
      ],
      "mode": "production",
      "output": Object {
        "filename": "index.jsbundle",
        "globalObject": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": true,
          "preloadBundles": Array [
            "base_dll",
          ],
          "sourceMap": true,
        },
        DllReferencePlugin {
          "options": Object {
            "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
            "manifest": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
            "sourceType": "this",
          },
        },
      ],
      "target": "webworker",
    },
  },
}
`;

exports[`makeConfig with multiple bundles should create config for packager server 1`] = `
Object {
  "bundles": Object {
    "app": Object {
      "app": true,
      "assetsDest": "",
      "dependsOn": Array [
        "base_dll",
      ],
      "dev": false,
      "dll": false,
      "entry": Array [
        "./spp.js",
      ],
      "external": false,
      "hasteOptions": Object {},
      "minify": false,
      "minifyOptions": undefined,
      "name": "app",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
    "base_dll": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [],
      "dev": false,
      "dll": true,
      "entry": Array [
        "./dll.js",
      ],
      "external": false,
      "hasteOptions": Object {},
      "minify": false,
      "minifyOptions": undefined,
      "name": "base_dll",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
    "index": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [
        "base_dll",
      ],
      "dev": false,
      "dll": false,
      "entry": Array [
        "./host.js",
      ],
      "external": false,
      "hasteOptions": Object {},
      "minify": false,
      "minifyOptions": undefined,
      "name": "index",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
  },
  "platforms": Array [
    "ios",
    "android",
  ],
  "server": Object {
    "host": "localhost",
    "port": 8081,
  },
  "templates": Object {
    "filename": Object {
      "__server__": "[bundleName].[platform].bundle",
      "android": "[bundleName].[platform].bundle",
      "ios": "[bundleName].jsbundle",
    },
  },
  "webpackConfigs": Object {
    "app": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Array [
        "./spp.js",
      ],
      "mode": "production",
      "output": Object {
        "filename": "app.ios.bundle",
        "globalObject": "this",
        "library": "app",
        "libraryTarget": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": false,
          "preloadBundles": Array [
            "base_dll",
          ],
          "sourceMap": true,
        },
        DllReferencePlugin {
          "options": Object {
            "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
            "manifest": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
            "sourceType": "this",
          },
        },
      ],
      "target": "webworker",
    },
    "base_dll": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Array [
        "./dll.js",
      ],
      "mode": "production",
      "output": Object {
        "filename": "base_dll.ios.bundle",
        "globalObject": "this",
        "library": "base_dll",
        "libraryTarget": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": false,
          "preloadBundles": Array [],
          "sourceMap": true,
        },
        DllPlugin {
          "options": Object {
            "name": "base_dll",
            "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
          },
        },
      ],
      "target": "webworker",
    },
    "index": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Array [
        "./host.js",
      ],
      "mode": "production",
      "output": Object {
        "filename": "index.ios.bundle",
        "globalObject": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": false,
          "preloadBundles": Array [
            "base_dll",
          ],
          "sourceMap": true,
        },
        DllReferencePlugin {
          "options": Object {
            "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
            "manifest": "<<REPLACED>>/packages/haul-core/src/preset/__tests__/base_dll.manifest.json",
            "sourceType": "this",
          },
        },
      ],
      "target": "webworker",
    },
  },
}
`;

exports[`makeConfig with single bundle should create config for basic-bundle 1`] = `
Object {
  "bundles": Object {
    "index": Object {
      "app": false,
      "assetsDest": "",
      "dependsOn": Array [],
      "dev": false,
      "dll": false,
      "entry": Array [
        "index.js",
      ],
      "external": false,
      "hasteOptions": Object {},
      "minify": false,
      "minifyOptions": undefined,
      "name": "index",
      "platform": "ios",
      "providesModuleNodeModules": Array [
        "react-native",
      ],
      "root": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "sourceMap": true,
      "type": "basic-bundle",
    },
  },
  "platforms": Array [
    "ios",
    "android",
  ],
  "server": Object {
    "host": "localhost",
    "port": 8081,
  },
  "templates": Object {
    "filename": Object {
      "__server__": "[bundleName].[platform].bundle",
      "android": "[bundleName].[platform].bundle",
      "ios": "[bundleName].jsbundle",
    },
  },
  "webpackConfigs": Object {
    "index": Object {
      "context": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
      "devtool": false,
      "entry": Array [
        "index.js",
      ],
      "mode": "production",
      "output": Object {
        "filename": "index.jsbundle",
        "globalObject": "this",
        "path": "<<REPLACED>>/packages/haul-core/src/preset/__tests__",
        "publicPath": "http://localhost:8081/",
      },
      "plugins": Array [
        DefinePlugin {
          "definitions": Object {
            "__DEV__": false,
            "process.env": Object {
              "NODE_ENV": "\\"production\\"",
            },
          },
        },
        LoaderOptionsPlugin {
          "options": Object {
            "debug": false,
            "minimize": false,
            "test": Object {
              "test": [Function],
            },
          },
        },
        SourceMapDevToolPlugin {
          "fallbackModuleFilenameTemplate": "webpack://[namespace]/[resourcePath]?[hash]",
          "moduleFilenameTemplate": "[absolute-resource-path]",
          "namespace": "",
          "options": Object {
            "filename": "[file].map",
            "module": true,
            "moduleFilenameTemplate": "[absolute-resource-path]",
            "test": /\\\\\\.\\(js\\|jsx\\|css\\|ts\\|tsx\\|\\(js\\)\\?bundle\\)\\(\\$\\|\\\\\\?\\)/i,
          },
          "sourceMapFilename": "[file].map",
          "sourceMappingURLComment": "
//# sourceMappingURL=[url]",
        },
        WebpackBasicBundlePlugin {
          "bundle": false,
          "preloadBundles": Array [],
          "sourceMap": true,
        },
      ],
      "target": "webworker",
    },
  },
}
`;
